RFC-0007-plan-constraints-grammar.md

```diff
@@ -0,0 +1,160 @@
+# **RFC007 for Presto**
+
+## Grammar for Plan Constraints
```

Comment:
can you add an overview of other engines that allow query hints and what their grammar looks like?

Reply:
The [overview doc](https://prestodb.io/wp-content/uploads/Search-Space-Improvements-Plan-Constraints.pdf) links out to how other engines implement plan constraints. I copied over the relevant info to the background section